Added expanded combos with buffs for Barbs

jayRyan24 opened the pull request:
Added the ability for Barbs to maintain the short-term buffs of Damage and Accuracy and then fill in with Flame to maintain high IF.

rpherbig commented on combat-trainer.lic:
```diff
@@ -1984,31 +2011,53 @@ class GameState
-    result = bput("analyze #{@analyze_combo_type}", 'Analyze what\?', 'by landing a .*')
-    waitrt?
+  def perform_analyze?(combo_type, expertise_requirement)
+    return false if combo_type == "accuracy" && !Flags["ct-#{combo_type}-ready"]
```
You could combine these two cases by checking `combo_type != 'flame' && ...`

rpherbig commented on combat-trainer.lic:
```diff
@@ -1314,6 +1314,11 @@ class AttackProcess
     pause
     waitrt?
 
+    if reget(20, 'increase the accuracy')
```
I generally try to avoid using reget because it's not reliable. Can we instead reset flags if we've finished the combo?

jayRyan24 replied:
This is tricky. Just finishing the array doesn't mean that it was successful, sometimes it inputs actions that get triggers are completed, but don't really. This happens when something dies and you gotta switch targets, it leaves, it's out of range, etc. I thought about a flag that checked if doing the combo, but you still need to check if that text was received to know you have done it. A way to make this work without it, is just to remove the reget and let it attempt to go through, if it finds that it's been done and the combo is on cooldown, it sets the flag to false. This is actually part of the backup checks to make sure it doesn't continuously loop. Removing reget will put this as the main check. The only issue is that it might appear spammy the first time it goes through to find out if it's on cooldown.

rpherbig commented on combat-trainer.lic:
```diff
@@ -1707,6 +1712,20 @@ class GameState
+    echo("  @use_barb_combos: #{@use_barb_combos}") if $debug_mode_ct
+
+    # For deprecation support
+    if !@analyze_combo_type.nil?
```
You could simplify this to `if @analyze_combo_type`

rpherbig commented on combat-trainer.lic:
```diff
@@ -1707,6 +1712,20 @@ class GameState
+
+    # For deprecation support
+    if !@analyze_combo_type.nil?
+      @use_barb_combos = true
```
I recommend doing this before echoing the value on line 1716 (or better yet, echo something else here so we know the setting got overriden).

jayRyan24 commented on combat-trainer.lic:
```diff
@@ -1912,6 +1931,14 @@ class GameState
     @aim_queue.shift
   end
 
+ def next_combo_action
```
Do you mean both the aim_queue.shift and the Analyze_combo_array? def next_combo_action would be a new function that I included to follow convention, I can just remove it from the commit.

rpherbig replied:
I was referring to `next_combo_action` and `use_barb_combos?`. They are only used internally to GameState.

The other methods you were looking at (with aim_queue.shift) are used externally.

jayRyan24 replied:
That's fine, I'll just remove it from the commit and PR.

jayRyan24 commented:
I had a lot of issues with my main PC and switched it over to a Linux box, etc. I deleted the old fork and made a new one. I never stopped using my old files though, and have been running them without issue for nearly a month. Please use the Pull request.
